pkg/auth/service.go

```diff
@@ -1902,13 +1903,44 @@ func NewAPIAuthService(apiEndpoint, token string, secretStore crypt.SecretStore,
 	} else {
 		cache = &DummyCache{}
 	}
-	res := &APIAuthService{
+	healthRes := &APIAuthService{
 		apiClient:   client,
 		secretStore: secretStore,
 		logger:      logger,
 		cache:       cache,
 	}
-	return res, nil
+
+	// Perform health check for API auth service
+	bo := backoff.NewExponentialBackOff()
+	bo.MaxInterval = 5 * time.Second
+	bo.InitialInterval = 1 * time.Second
+	bo.MaxElapsedTime = 20 * time.Second
```

Comment:
Can you make this configurable with defaults? 
When debugging such issues that would come in handy to extend the retries for example to keep the Pods up while sshing into the pod and debugging.

Reply:
I'll add a configuration parameter for the max elapsed time and make the rest constants